File: lightbulb-accordionview/src/main/java/com/github/RooneyAndShadows/lightbulb/accordionview/animation/AccordionShowHideAnimation.java
package com.github.rooneyandshadows.lightbulb.accordionview.animation;

import android.view.View;

public class AccordionShowHideAnimation implements AccordionAnimation {
    private final View targetView;

    public AccordionShowHideAnimation(View targetView) {
        this.targetView = targetView;
    }

    @Override
    public void expand(int duration) {
        targetView.setVisibility(View.VISIBLE);
    }

    @Override
    public void collapse(int duration) {
        targetView.setVisibility(View.GONE);
    }

    public boolean hasRunningAnimation() {
        return false;
    }
}

File: lightbulb-accordionview/src/main/java/com/github/RooneyAndShadowss/lightbulb/accordionview/animation/AccordionShowHideAnimation.kt
package com.github.rooneyandshadows.lightbulb.accordionview.animation

import android.view.View

class AccordionShowHideAnimation(private val targetView: View?) : AccordionAnimation {
    override fun expand(duration: Int) {
        targetView!!.visibility = View.VISIBLE
    }

    override fun collapse(duration: Int) {
        targetView!!.visibility = View.GONE
    }

    override fun hasRunningAnimation(): Boolean {
        return false
    }
}